### PR TITLE
Routed queries

### PR DESCRIPTION
**Pull request by danielnaab**
This adds support for search.gov routed queries.

If the routed query is a FAQ search page, additional search behavior as defined in #736 is implemented.

Due to a bug in the autocomplete widget, we can not currently pre-fill the search terms. However, there is a fix available that I will monitor and deploy asap (see #766).

Please confirm the following steps are completed:

* [x] Choose the appropriate target branch:
  * Content
    * `preview` (approved content) <- *Content branch*
    * `master` (production) <- `preview`
* [ ] Assign an appropriate reviewer:
  * *Content admin* or *Project lead* for merge to `preview` (approved content)
  * *Engineer* for merge to master (production)

:sunglasses: [PREVIEW URL](https://cg-dae9433e-23a0-46d7-bafb-dc79d2d3756c.app.cloud.gov/preview/18f/cv_faq/no-results-text/)

### CODE REVIEW

**Comment by jonadecker** [comment]
@danielnaab I've tested each of the routed queries and it appears to be constructing the query correctly but generating the 404 page. Sent examples via Slack.

**Comment by danielnaab** [comment]
> @danielnaab I've tested each of the routed queries and it appears to be constructing the query correctly but generating the 404 page. Sent examples via Slack.

@jonadecker the Federalist base URL is now prefixed on the routed search URLs, so testable now. I'm still working to address issues caused by prefilling the current search terms in the input box.

File: _assets/js/search.js [hunk 1]
```diff
@@ -21,19 +21,39 @@
...
+        const url = new URL(response.route_to);
+        if (url.host == 'faq.coronavirus.gov' && url.pathname == '/search/') {
+          // Return the URL minus the hostname, with the orginal query included
+          return `${BASE_URL}${url.pathname}${url.search}&source=${encodeURIComponent(query)}`;
+        }
+
+        // This is a routed query to something other than search results.
+        return response.route_to;
+      })(),
+      results: response.web && response.web.results ? response.web.results.map(result => {
```
Comment (piascikj) [inline _assets/js/search.js:49]: Could be written...
```javascript
response.web.results.map(({url, title, description}) => ({url, title, description}));
```

File: _assets/js/search.js [hunk 2]
```diff
@@ -21,19 +21,39 @@
...
+        if (!response.route_to) {
+          return null;
+        }
+
+        // If this is a routed query pointing to production search results, extract
+        // search parameters out.
+        // So we can test lower environments against the production search.gov,
+        // assume production hostname.
+        const url = new URL(response.route_to);
+        if (url.host == 'faq.coronavirus.gov' && url.pathname == '/search/') {
```
Comment (piascikj) [inline _assets/js/search.js:41]: Use `===` to compare strings

File: _assets/js/search.js (hunk 1 above)
Comment (danielnaab) [reply _assets/js/search.js:49]: True.. although `snippet` is the source for `description` here, so I think I'll keep it for explicitness.

Comment (piascikj) [reply _assets/js/search.js:49]: gotcha!

**Pull request merged**